[Lee En Hao] iP

LeeEnHao opened the pull request.

Commit:
Add Gradle support

LeeEnHao committed:
Batman! This commit has no parents!

LeeEnHao committed:
Level-1 Commit: Echo functionality and basic file structure.

LeeEnHao committed:
Added:

dependencies/executor/Executor.java - Class to execute commands given by
user.
dependencies/storage/Store.java - Class to act as a database for todo
list.

LeeEnHao committed:
Added a serializable class DB.java in dependencies/storage.

LeeEnHao committed:
Added a temporary function `level2Exec(String command) to satisfy the
level 2 requirements.

LeeEnHao committed:
Late night commit: level2Exec function update

LeeEnHao committed:
Added files: dukeexceptions,
Checker.java for checking validity of commands
DK.java for printing to std:out
Refined the behaviour of how commands are parsed and excuted.

LeeEnHao committed:
Added Task classes representing a task on the todolist.

LeeEnHao committed:
Added Events, ToDos and Deadlines, which are all subclasses of Task.
This categorises the task on the todolist into different groups of
tasks.

LeeEnHao committed:
Added toString() methods for all three subclasses of Task. Should be
able to render properly as a string in std:out.

LeeEnHao committed:
Fixed typo in Deadlines.java toString().

LeeEnHao committed:
Updated and added a new interface to abstract out the commands.

LeeEnHao committed:
Added a new class Command.

LeeEnHao committed:
Further adstraction of commands into a command object. Refactored the
classes using/interacting with Commands. Made Task an abstract class.

LeeEnHao committed:
Fixed separator indentation

LeeEnHao committed:
Resolved issues within Checker. Program now compiles and runs. Issues
yet to be resolved:
1) Listing size of todolist.
2) Catching errorneous commands.
3) Deletion of tasks.
4) Fully separate Done from Tasks.

LeeEnHao committed:
Added methods getListSize and getNumOfCompleted to Store.java. Changed
the array storage to Arraylist in Store.java.

LeeEnHao committed:
Refined the presentation of messages by Duke by including reminders of
number of tasks in the list/number of completed tasks/incomplete tasks.

LeeEnHao committed:
Updated /.gitignore file for text-ui-test.

LeeEnHao committed:
Updated the javac filepath

LeeEnHao committed:
Added and updated the EXPECTEC.TXT file.

LeeEnHao committed:
Updated the /.gitignore file

LeeEnHao committed:
Added new static method to Command.java: creates a "delete" command.
Enables user to delete a task from a todolist.

LeeEnHao committed:
Added the ability for Duke to delete tasks.

LeeEnHao committed:
Resolved the indexing error for deleting tasks.

LeeEnHao committed:
Updates /.gitignore file ACTUAL.txt to ACTUAL.TXT

LeeEnHao committed:
Changed enums CommandType from public to default access

LeeEnHao committed:
Add DukeExceptions

LeeEnHao committed:
Add private method to handle the parsing of string input by user.

This removed the need to handle the parsing in the method itself. Allows
more room to explore other methods of using regex to check for implicit
passing fo tasks to Duke.

E.g. "event meeting /at Monday" - explicit passing of input.
     "I have a meeting at Monday" - implicit passing of input.

Note: Changed access of enums CommandType to public for Executor.java to
access.

LeeEnHao committed:
Add exceptions to be thrown if command given is invalid or does not
follow the format. Exceptions thrown by Checker is caught in Parser.
Prevents invalid arguments from being passed to Executor and Storage.

LeeEnHao committed:
Add Unspecified date exception to handle events and deadlines without a
specified date.

LeeEnHao committed:
Throw EmptyTaskexception id "done" command has no follow up task number.

LeeEnHao committed:
Rename DK.java to Ui.java

LeeEnHao committed:
Merge branch 'ui-text-test' into master

LeeEnHao committed:
Reorganise methods in Checker.java

LeeEnHao committed:
Fix typos in comments

LeeEnHao committed:
Made Task class serializable

LeeEnHao committed:
Add java.nio.file imports to Store.java to enable file manipulation

LeeEnHao committed:
Delete DB.java

LeeEnHao committed:
Add interface Schedulable

Separates the Task implementation from being handled directly by the
storage to improve extensibility.

LeeEnHao committed:
Add reading and writing to taskdata.txt file

Store able to read and write to file to save data.i

LeeEnHao committed:
Add MissingListExecption for missing save file

LeeEnHao committed:
Add Loader class to handle the reading and writing to files to save
data

LeeEnHao committed:
Fix createEvent in checker to createDeadline

Original: createEvent was being called for deadlines.
Fixed: createDeadline is now being called for deadlines.

LeeEnHao committed:
Update runtest.bat

LeeEnHao committed:
Update gitignore to include data/taskdata.txt

LeeEnHao committed:
Update runtest.bat file

LeeEnHao committed:
Update Javadocs for TaskDate

LeeEnHao committed:
Update Task constructors for Tasks with a deadline/date

LeeEnHao committed:
Add hasADate checker function for Tasks to check for presence of TaskDate objects

LeeEnHao committed:
Update Checker and Parser to handle different exceptions

Parser: Catches all possible DukeExceptions and returns a reply to the user accordingly.
Checker: Throws InvalidDateException for date strings that cannot be parsed. Be it due to wrong format or wrong impossible dates.

LeeEnHao committed:
Merge branch 'branch-Level-8' into master

KenGondor committed:
Change Store.java to TaskList.java and Loader.java to Storage.java and
added javadocs for DukeExceptions

KenGondor committed:
Nit: add description for 's' param of Ui.java takeCommand(String s)
method

KenGondor committed:
Add packages for main/java

KenGondor committed:
Fix TaskDate.isValidFormat(String date) to check for valid format of dates

KenGondor committed:
Fix main files

KenGondor committed:
Refactored code to provide a more clean error handling of bad user input

TaskDate.java: Implemented Serializable.
TaskList.java: removeed Serialiazable (Flip-flip commit).
text-ui-test: Updated EXPECTED.TXT and added cmd to runtest.bat to
delete data cache.
Storage.java: Commented out messages printing to System.out and added
new messages to give a more clean feel. Additionally, removed the
exception print stack trace.

KenGondor committed:
Add package for Duke and Ui. Add Javadocs for FNV class for storage
module

KenGondor committed:
Changed structure of project and packages

KenGondor committed:
Refactored code in runtest.bat

KenGondor committed:
Add Junit 5 dukeexceptions test drivers in test/

KenGondor committed:
Add test for TaskDateTest, CommandTest, and Executor test

KenGondor committed:
Change to multiple single import statements for Executor TaskDate

KenGondor committed:
Nit: Minor spacing

KenGondor committed:
Add Javadocs for Task.java static factory methods and public methods

KenGondor committed:
Add

KenGondor committed:
Revert "Add"

This reverts commit 76f3aa52a4af6869f9b3f0943f020dd2aa9b2e56.

KenGondor committed:
Nit: fix typos in Task.java javadocs

KenGondor committed:
Add TaskTest.java and changes getDateLine method to getDateString

KenGondor committed:
Add CheckerTest for valid commands

KenGondor committed:
Add ParserTest

KenGondor committed:
A-Jar

KenGondor committed:
Add and reformated Javadocs to comply with Javadoc style

KenGondor committed:
Change method names to comply with coding conventions

KenGondor committed:
Add Find functionality to find matching tasks

KenGondor committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

KenGondor committed:
Nit: Fix typos and redundant assignments

KenGondor committed:
Merge branch 'branch-Level-9' into branch-A-CodingStandard

Resolved merge conflict in Task.java and Executor.java
Task.java: change of method names
Executor.java: Nit-picking of comment lines

KenGondor committed:
Nit: Javadocs of findMatching in TaskList.java

KenGondor committed:
Add Javadocs for interface: Schedulable

KenGondor committed:
Rename Parser to Controller and Checker to Parser

KenGondor committed:
Nit: Fix Command.java createFindCommand Javadocs

KenGondor committed:
Refactor Controller object in Ui.java

KenGondor committed:
Add three static factory methods for specific ADD commands

KenGondor committed:
Refactor code in Parser, Executor, Controller and Command modules

Reason: Executor class is returning a parsed response to the user. It
should focus on executing the logical commands on the TaskList only.

Controller now explicitly stringifies a response based on each command
and returns it to the Ui. The response is a reply that shows what was
done by the Executor in carrying out the command.

Command no longer has methods to create different kinds of ADD commands:
EVENT, DEADLINE and TODO. Instead all can be grouped under ADD since the
response is the same: adding to the TaskList.

Nit: Fix style in Parser and added comments.

KenGondor committed:
Remove unused imports and fields

KenGondor committed:
Nit: Remove commented out fields

KenGondor committed:
Merge branch 'master' into add-gradle-support

LeeEnHao committed:
Nit: Adjust spacing of dividers

LeeEnHao committed:
Merge branch local 'better-abstraction' into master

Provides better abstraction from executor and parser.

LeeEnHao committed:
Change reply for wrong date format in Controller.java

LeeEnHao committed:
Nit: replace for loop with enhanced for loop in TaskList module

LeeEnHao committed:
Nit: remove unused enums in CommandType

LeeEnHao committed:
Added JavaFX library in dependencies

LeeEnHao committed:
Reorganise coding style to follow Code Conventions

LeeEnHao committed:
Implement regex word boundaries to check for commands

Previous implementation allows the command to be nested in another word.

LeeEnHao committed:
Add password setting feature

LeeEnHao committed:
Add Javadoc comments

LeeEnHao committed:
Add test for Parser.java

LeeEnHao committed:
Add varargs and Javadoc for Parser

LeeEnHao committed:
Delete pw.dat

LeeEnHao committed:
Update gitignore

LeeEnHao committed:
Add checkstyle and remove unused imports in Executor

LeeEnHao committed:
Add Javafx main stage to duke main application

Duke now extends Application. More to be done.

LeeEnHao committed:
Add javafx launcher

LeeEnHao committed:
Nit: Remove unused import

LeeEnHao committed:
Add styling to GUI scene components

LeeEnHao committed:
Level-10: Iteration 1

LeeEnHao committed:
Add scroll bar auto scroll to bottom

sigmund-c commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,112 @@
+        primaryStage.setMinWidth(400.0);
+        primaryStage.setMinHeight(600.0);
+
+        // Create button with text "Send" as the send button
+        sendButton = new Button();
+        sendButton.setText("Send");
+
+        // Create the scroll pane
+        scrollPane = new ScrollPane();
+
+        // Create the text input
+        userInputTextField = new TextField();
+
+        // Create the vertical box pane for the dialogue container
+        dialogueContainer = new VBox();
+
+        // Add the content to the scroll pane for a scrollable content
+        scrollPane.setContent(dialogueContainer);
+
+        // Creates the main layout pane
+        AnchorPane mainLayout = new AnchorPane();
+        mainLayout.getChildren().addAll(scrollPane, userInputTextField, sendButton);
+
+        // Sets the main layout as a scene
+        Scene scene = new Scene(mainLayout, 300 ,500);
+
+        // Sets the scene for the stage
```
The comments might be a bit redundant since the code already somewhat tells what is happening? But not sure.

royleochan replied:
I feel the comments might help make things clearer especially for readers who are new to javaFX.

sigmund-c commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,74 @@
+ *
+ */
+class Ui {
+    private static final String DIVIDER = "____________________________________________________________\n";
+    private static final String GREETING = "Hello, I'm Duke\nwhat can I do for you?\n";
+    private static final String LOGO = " ____        _        \n"
+                                    + "|  _ \\ _   _| | _____ \n"
+                                    + "| | | | | | | |/ / _ \\\n"
+                                    + "| |_| | |_| |   <  __/\n"
+                                    + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String CIAO = DIVIDER + "Spero di rivederti presto\n" + DIVIDER;
+    private static final String CONVO_START = DIVIDER + LOGO + "\n" + GREETING + DIVIDER;
+    private static final String END = "end|ciao|bye|close|exit|nights|shutdown";
```
Since these constants are associated, it might make sense to prefix them with a common name (eg. MSG_DIVIDER, MSG_GREETING, ...).

sigmund-c commented on src/main/java/duke/dependencies/executor/Executor.java:
```diff
@@ -0,0 +1,152 @@
+     */
+    public String receiveAndExec(Executable executable) {
+        // TODO: Adding of new commands is to be done here.
+        switch(executable.getType()) {
+            case LIST:
+                setState(LIST);
+                break;
+
+            case ADD:
+                setState(ADD);
+                break;
+
+            case DELETE:
+                setState(DELETE);
+                break;
+
+            case DONE:
+                setState(DONE);
+                break;
+
+            case FIND:
+                setState(FIND);
+                break;
+
+            case CLEAR:
+                setState(CLEAR);
+                break;
+
+            default:
+                setState(INVALID);  // Should never reached this stage.
+                break;
+
```
Another minor one, but the coding standard mentions that there's no need to indent for the `case` clauses.

royleochan replied:
Agreed, no indentation needed for case clauses. To add on, I think there isn't supposed to be a line break between cases as well (referring to the code snippet in the coding standards) but i'm not sure if that must be followed strictly.

sigmund-c commented on src/main/java/duke/dependencies/task/Task.java:
```diff
@@ -0,0 +1,241 @@
+        @Override
+        public String toString() {
+            return String.format(
+                    "[Event][%s] %s (at: %s)", super.state == UNFINISHED ? "X"
```
Perhaps the last `: "✓",`  line can be in the same line as the previous to improve readability?

royleochan replied:
Agreed, unless the character line limit is reached.

sigmund-c commented on src/main/java/duke/dependencies/storage/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+    public String getTodosInList() {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < todoList.size(); i++) {
+            sb.append(i+1);
```
Mentioning the reasoning behind the +1 might be a good idea

sigmund-c commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,112 @@
+        }
+
+        /* Main App Loop */
+        int x = 1;
```
Consider naming x as a more obvious name (maybe returnValue?) just to make it clearer.

royleochan replied:
Agreed, even though the scope is small it might be clearer to use a more descriptive variable name. Looking at your takeInputAndReturn method maybe x can be renamed to commandStatus?

sigmund-c reviewed (commented):
Most of the comments I have are the more nitpicky ones with the format. Otherwise, the code is pretty well done with and organized.

royleochan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,112 @@
+     *                     the application scene can be set.
+     *                     Applications may create other stages, if needed, but they will not be
+     *                     primary stages.
+     * @throws Exception if something goes wrong
```
Can consider being replacing "if something goes wrong" in the javadoc to some thing more specific. I'm not sure what because I haven't done the java FX portion but if its meant to be generic then its fine!

royleochan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,74 @@
+        System.out.println(CONVO_START);
+    }
+
+    public boolean checkAuth() {
```
Can consider writing java docs for these methods (even though its trivial) because based on the coding standards all public class and methods should have headers. I noticed the same issue with some of the other public classes as well.

royleochan commented on src/main/java/duke/dependencies/parser/Controller.java:
```diff
@@ -0,0 +1,148 @@
+        if (!isInUserAuthenticationMode) {
+            try {
+                parser = Parser.parseAndCheck(command);
+            } catch (EmptyTaskException e) {
```
Can consider removing the line breaks in the catch blocks as the code snippet in the coding standards for try-catch statements don't have the line breaks.

royleochan commented on src/main/java/duke/dependencies/parser/Controller.java:
```diff
@@ -0,0 +1,148 @@
+            /* If no DukeException is thrown, it means user input is valid and follows all specific format */
+
+            Executable e = parser.getExecutable();
+
```
Should the line breaks in the variable declarations here be removed?

royleochan commented on src/main/java/duke/dependencies/parser/Parser.java:
```diff
@@ -0,0 +1,223 @@
+
+
+    /**
+     *
```
I think a brief description for each method is needed at the start of the javadocs (before @param, @return, @throws)

royleochan commented on src/main/java/duke/dependencies/parser/Parser.java:
```diff
@@ -0,0 +1,223 @@
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
```
Think you might have missed out javadocs for the class itself.

royleochan commented on src/main/java/duke/dependencies/task/Task.java:
```diff
@@ -0,0 +1,241 @@
+
+import java.io.Serializable;
+
+import static duke.dependencies.task.CompletionState.*;
```
Think the imports should be specified instead of using *

royleochan commented on src/main/java/duke/dependencies/task/TaskDate.java:
```diff
@@ -0,0 +1,115 @@
+        return date.format(DateTimeFormatter.ofPattern("MMM dd uuuu"));
+    }
+
+
```
Maybe these comments can be removed? Unless its still needed for reference.

LeeEnHao committed:
Add GUI user pic

LeeEnHao committed:
Add images

LeeEnHao commented:
Thanks yall for the feedback, I guessed I took many shortcuts in doing this. I have committed most changes recommended into the current branch and will merge it into the master soon. The exception being the MSG_DIVIDER and string constants, that being a personal choice since it is all included in the class and not used anywhere else. However I have added comments/oneline docs to the constants to clarify their use for future reference.

LeeEnHao committed:
Add gui interaction with user

Major issues to be fix: prettifying the output for duke.

LeeEnHao committed:
Remove deprecated formatting for CLI to adapt to gui

LeeEnHao committed:
Add reformatted messages

LeeEnHao committed:
Add assertions at TaskList, method add(Task task)

Checks if the task added is not a Misc. task or empty task. Reason for
change: Empty or misc task are not task that the user would want to add
to the task list.

LeeEnHao committed:
Refactor user details authentication

Moved user details checking to a separate class to implement the validation
of the user details.

LeeEnHao committed:
Merge pull request #2 from LeeEnHao/branch-A-CodeQuality

Branch a code quality

LeeEnHao committed:
Add C-MassOps and A-Streams

LeeEnHao committed:
Merge pull request #1 from LeeEnHao/branch-A-Assertions

Branch a assertions

LeeEnHao committed:
Merge branch 'master' into BCD-Extension

LeeEnHao committed:
Merge pull request #3 from LeeEnHao/BCD-Extension

Add C-MassOps and A-Streams

LeeEnHao committed:
Add lambdas for Predicate in the filtering of the tasklist stream

LeeEnHao committed:
Merge pull request #4 from LeeEnHao/branch-A-Streams

Add lambdas for Predicate in the filtering of the tasklist stream

LeeEnHao committed:
Add final event handling for password authentication

LeeEnHao committed:
Add better GUI and some styling for stage title and icon

LeeEnHao committed:
Add changes to textfield effects

LeeEnHao committed:
Add checks for greetings and modified GUI of dialog box

LeeEnHao committed:
Change to pipboy

LeeEnHao committed:
Add simple help message and story

LeeEnHao committed:
Add png images

LeeEnHao committed:
Rename png files

LeeEnHao committed:
Update README.md

LeeEnHao committed:
Update UG

LeeEnHao committed:
Update UG
